### PR TITLE
Feat PLA-6929 apply cop

### PR DESCRIPTION
**Pull request by wojtiku**
### Issue link

https://pitcher-ag.atlassian.net/browse/PLA-6929

### 📖 Description

This pull request applies the rules agreed on Front-end CoP. Apart from that it changes what we use for sorting imports in the hope it will play more nicely with typescript.

Application of the rules looks like that: https://github.com/PitcherAG/salesforce-demo-ui/pull/152

### 📷 Screenshots
n/a

### CODE REVIEW

File: javascript.js [hunk 1]
```diff
@@ -8,16 +8,30 @@ const pitcherConfig = require('./pitcher.config')
...
+    },
+  },
+  rules: {
+    'import/order': [
+      'error', {
+        groups: ['builtin', 'external', 'internal', 'parent', 'sibling', 'index', 'object', 'type'],
+        pathGroups: [
+          {
+            pattern: '@/**',
+            group: 'external',
```
Comment (AykutBurak) [inline javascript.js]: asking because I don't really know: 
shouldn't this be internal?

File: javascript.js [hunk 2]
```diff
@@ -8,16 +8,30 @@ const pitcherConfig = require('./pitcher.config')
...
+            position: 'after'
+          }
+        ],
+        pathGroupsExcludedImportTypes: ['builtin']
+      }
     ],
+    "import/newline-after-import": ['error', { count: 1 }],
+    'node/file-extension-in-import': ['error', 'always', { '.js': 'never', '.vue': 'never', '.ts': 'never' }],
```
Comment (AykutBurak) [inline javascript.js:22]: finally 😄

File: vue.js [hunk 3]
```diff
@@ -25,7 +25,12 @@ module.exports = {
     'prettier/prettier': ['error', prettierConfig],
 
     /* Naming rules*/
-    'vue/custom-event-name-casing': ['error'],
+    'vue/custom-event-name-casing': ["error",
+      "camelCase",
+      {
+        "ignores": []
+      }
+    ],
```
Comment (AykutBurak) [inline vue.js:33]: FINALLY!!! 😄

**Review by AykutBurak — DISMISSED** [review]

**Review by pegiadise — DISMISSED** [review]

File: javascript.js [hunk 4]
```diff
@@ -8,16 +8,24 @@ const pitcherConfig = require('./pitcher.config')
...
+            pattern: '@/**',
+            group: 'external',
+            position: 'after',
+          },
+        ],
+        pathGroupsExcludedImportTypes: ['builtin'],
       },
     ],
+    'import/newline-after-import': ['error', { count: 1 }],
+    'node/file-extension-in-import': ['error', 'always', { '.js': 'never', '.vue': 'never', '.ts': 'never' }],
```
Comment (kouts) [inline javascript.js:22]: Has this been tested with Vite? I thought Vite doesn't allow extension-less Vue imports.
https://github.com/vitejs/vite/issues/178

File: javascript.js (hunk 1 above)
Comment (wojtiku) [reply javascript.js]: This is internal, yeah. But we put this AFTER the external group :D

File: javascript.js (hunk 4 above)
Comment (wojtiku) [reply javascript.js:22]: We stopped using Vite for now because things like this one. I think we will revisit when it is more mature :)

**Review by AykutBurak — DISMISSED** [review]

**Comment by wojtiku** [comment]
Actually this sorting plugin is not the best TBH :(. It is not consistent enough and has some quirks. It needs a bit more time.

**Comment by wojtiku** [comment]
In the end I reverted the sorting changes.

**Review by AykutBurak — APPROVED** [review]

**Review by sermetege — APPROVED** [review]

**Pull request merged**